doc/iterations/2023.6/shapes/sendMessageToFrontend.adoc

```diff
@@ -0,0 +1,59 @@
+= (L) Display feedback message on the frontend
+
+== Problem
+
+It's not possible for a specifier to alert end user after a tool action.
+
+== Key Result
+
+The specifier can call an API in his java service to display a message on the UI.
+The representation of the messages can be specified to alert the end user.
+
+== Solution
+
+Offer an API to display messages on the UI.
+
+[source,java]
+----
+interface IFeedbackMessageService {
+    void addFeedbackMessage(String message, FeedbackLevel level);
+    void addFeedbackMessage(String message, FeedbackStyle style);
```

Comment:
Can you explain why having two methods?
What's the difference betwee `FeedbackLevel` and `FeedbackStyle`?
My interpretation would be that `FeedbackStyle` allows fine-grained customization while `FeedbackLevel` has pre-defined values which correspond to common styles for convenience.

If that's the case, `FeedbackLevel` could probably be replaced by pre-defined constant values for `FeedbackStyle`:

```java
    record FeedbackStyle (int fontSize, String labelColor, String borderColor, String backgroundColor) {
        public static final FeedbackStyle INFO = new FeedbackStyle(16, "black", "green", "lightgreen");
        public static final FeedbackStyle WARNING = new FeedbackStyle(16, "black", "darkorange", "orange");
        public static final FeedbackStyle ERROR = new FeedbackStyle(16, "white", "darkred", "red");
    }
```

If that's not the case, and the level is really something separate from just a visual style, then I don't understand why we could set only one for a message.

Reply:
The screenshot below shows icons in addition to the text, so maybe FeedbackLevel is about which icon to display? It's not mentioned.

Reply:
The enum was designed to directly match the existing UI library behavior, while the other method offers the possiblity to create a custom representation.
But for this first implementation, we should probably offer the only option to choose between the three default behaviors.